### DISCUSSION TITLE
Show multiple vertical images on one screen

### DISCUSSION THREAD

**Discussion by desmond206x** [damongolding/immich-kiosk#17] [Feature request]
First of all, thanks for the good work!

I would like to have multiple images on one slide if the images are all vertical aligned. 
It should chance some of you logic, because now you have multiple timestamps from the images at the same time, but it would be nice to configure if the images will be selected randomly or maybe immich-kiosk will look for two or three images as imminant as possible.

**Comment by jorbig** [comment]
I second this request. Google's ambient mode already does this for multiple years:
https://www.androidpolice.com/2018/10/24/chromecast-smart-displays-start-showing-portrait-photos-side-side/
It's one of the reasons I haven't switched to Immich just yet.

**Comment by fillwe** [comment]
Would really like to see this as well! Only thing i'm missing from my Google Nest Hub right now

**Comment by damongolding** [comment]
Making some progress on this.

![localhost_3000__theme=solid show_time=false album=51be319b-55ea-40b0-83b7-27ac0a0d84a3 (1)](https://github.com/user-attachments/assets/e1715c57-a0da-4338-94e3-337e79478629)

with `image_fit=cover`

![localhost_3000__theme=solid show_time=false album=51be319b-55ea-40b0-83b7-27ac0a0d84a3 image_fit=cover](https://github.com/user-attachments/assets/8bfa81c1-981e-469b-8298-b4c70c788aef)

Same as above but using `theme=fade` (in current development)

![localhost_3000__theme=fade show_time=false album=51be319b-55ea-40b0-83b7-27ac0a0d84a3 image_fit=cover refresh=5 (1)](https://github.com/user-attachments/assets/688ac658-01ed-441f-acd6-894f3a29f435)

**Reply by desmond206x** [reply]
Love it, thank you

**Reply by damongolding** [reply]
In your mind are the two images from the same source i.e. same person or from the same album or a mix of sources (if multiple are set)?

**Reply by desmond206x** [reply]
Mh. Maybe allow both per config flag?
For my children I would like to mix both sources.

**Comment by damongolding** [comment]
I think i will go with random for now and if there is enough interest I can add that later

**Comment by damongolding** [comment]
If anyone is interested in being a beta tester they can use the docker image `damongolding/immich-kiosk-development:0.11.0` to test out the `layout=splitview`feature. If you do so and have any feedback you can either leave it here or hit me up on the Immich Discord server. There are some docs on it here [https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#layouts](https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#layouts)

**Reply by chris-burrow-apps** [reply]
I really like this split screen view. 

Feedback:
Maybe have it configurable like on google if possible.
- layout=splitview
- layout=splitviewportait (only enabled split screen when 2 portrait photos)

**Reply by desmond206x** [reply]
I am currently testing it, thanks
Works very good and responsive, but not exactly as I thought it would be.

You now have to switch the layout for all photos.
I would like to have horizontal images as single image view and if the next image is vertical, than I would like to show two vertical images to fit the most of the space on my 16:9 display.

But maybe I am alone with this request.

**Reply by damongolding** [reply]
> I am currently testing it, thanks
> 
> Works very good and responsive, but not exactly as I thought it would be.
> 
> 
> 
> You now have to switch the layout for all photos.
> 
> I would like to have horizontal images as single image view and if the next image is vertical, than I would like to show two vertical images to fit the most of the space on my 16:9 display.
> 
> 
> 
> But maybe I am alone with this request.

I think that's a very fair expectation and I will work towards it working that way. It will require a fair but more logic to get there. Thank you for the feedback 🙏

**Comment by damongolding** [comment]
I have made some improvements to Kiosk v0.11.0 with regards to 'splitview'. @desmond206x if you would be willing to pull the docker image again (v0.11.0) you should see it behave more like you would expect.

docs in the same place as before [https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#layouts](https://github.com/damongolding/immich-kiosk/tree/task/release?tab=readme-ov-file#layouts)

**Reply by desmond206x** [reply]
That's working fine.
Two images if vertical and one if horizontal.

Thank you very much!

**Reply by damongolding** [reply]
> That's working fine.
> 
> Two images if vertical and one if horizontal.
> 
> 
> 
> Thank you very much!

Good stuff! It was a surprising amount of work for such a small adjustment!

**Reply by dimatx** [reply]
I also just tested it, and it works great. Immich Kiosk has now enabled an Android device to seamlessly replace the slideshow formerly served from a Google Home display (then, using Google Photos) which failed recently. Thanks!

**Reply by AlenHodnik** [reply]
Can you call splitview layout directly with http url like http://ip:3000/image?layout=splitview ? Would that be possible?

**Reply by damongolding** [reply]
> Can you call splitview layout directly with http url like http://ip:3000/image?layout=splitview ? Would that be possible?

That wont work I'm afraid. The "/image" endpoint will only retrieve one image and serve that. The splitview layout is build using HTML.

**Comment by damongolding** [comment]
Implemented in 0.11.0
